src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useContext, useMemo, useState } from "react";
+
+// Components
+import {
+  ChatContext,
+  PrivateChatMessage,
+} from "components/context/ChatContext";
+import UserProfilePicture from "components/molecules/UserProfilePicture";
+import ChatBox from "components/molecules/Chatbox";
+
+// Hooks
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+
+// Utils | Settings | Constants
+import { DEFAULT_PARTY_NAME, VENUE_CHAT_AGE_DAYS } from "settings";
+import { getDaysAgoInSeconds } from "utils/time";
+import { WithId } from "utils/id";
+
+// Typings
+import { User } from "types/User";
+
+// Styles
+import "./ChatsList.scss";
+import { setPrivateChatMessageIsRead } from "components/organisms/PrivateChatModal/helpers";
+import { isEmpty } from "lodash";
+
+interface LastMessageByUser {
+  [userId: string]: PrivateChatMessage;
+}
+
+const ChatsList: React.FunctionComponent = () => {
+  const { user } = useUser();
+  const { privateChats, chatUsers } = useSelector((state) => ({
+    privateChats: state.firestore.ordered.privatechats,
+    chatUsers: state.firestore.data.chatUsers,
+  }));
+
+  const [selectedUser, setSelectedUser] = useState<WithId<User>>();
+
+  const discussionPartnerWithLastMessageExchanged =
+    privateChats &&
+    privateChats.reduce<LastMessageByUser>((agg, item) => {
+      let lastMessageTimeStamp;
+      let discussionPartner;
+      if (item.from === user?.uid) {
+        discussionPartner = item.to;
+      } else {
+        discussionPartner = item.from;
+      }
+      try {
+        lastMessageTimeStamp = agg[discussionPartner].ts_utc;
+        if (lastMessageTimeStamp < item.ts_utc) {
+          agg[discussionPartner] = item;
+        }
+      } catch {
+        agg[discussionPartner] = item;
+      }
+      return agg;
+    }, {});
+
+  const onClickOnSender = (sender: WithId<User>) => {
+    const chatsToUpdate = privateChats.filter(
+      (chat) => !chat.isRead && chat.from === sender.id
+    );
+    chatsToUpdate.map(
+      (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+    );
+    setSelectedUser(sender);
+  };
+
+  const roundToNearestHour = (seconds: number) => {
+    const oneHour = 60 * 60;
+    return Math.floor(seconds / oneHour) * oneHour;
+  };
```

Comment:
This could potentially be put with our other time helpers

- https://github.com/sparkletown/sparkle/blob/staging/src/utils/time.js

Since it's a `.js` file, we could either convert it now (probably OOS), or make a `timeTs.ts` file for now, and backlog it as tech debt to convert the old `time.js` file across to it then rename as `time.ts`

Reply:
fixed in 77fc5d6d1829a42da2152d00871cde0b6f47269d with `src/utils/time.js`